src/main/java/org/craftercms/engine/service/context/SiteContextManager.java

```diff
@@ -129,93 +128,43 @@ public class SiteContextManager implements ApplicationContextAware, DisposableBe
-    }
-
-    @Override
-    public void setApplicationContext(ApplicationContext applicationContext) {
-        this.applicationContext = applicationContext;
-    }
 
-    @Required
-    public void setContextFactory(SiteContextFactory contextFactory) {
         this.contextFactory = contextFactory;
-    }
-
-    @Required
-    public void setFallbackContextFactory(SiteContextFactory fallbackContextFactory) {
         this.fallbackContextFactory = fallbackContextFactory;
-    }
-
-    @Required
-    public void setSiteListResolver(final SiteListResolver siteListResolver) {
         this.siteListResolver = siteListResolver;
-    }
-
-    @Autowired
-    public void setEntitlementValidator(@Lazy final EntitlementValidator entitlementValidator) {
-        this.entitlementValidator = entitlementValidator;
-    }
-
-    @Required
-    public void setWaitForContextInit(boolean waitForContextInit) {
         this.waitForContextInit = waitForContextInit;
-    }
-
-    @Required
-    public void setJobThreadPoolExecutor(Executor jobThreadPoolExecutor) {
         this.jobThreadPoolExecutor = jobThreadPoolExecutor;
-    }
-
-    @Required
-    public void setDefaultSiteName(final String defaultSiteName) {
         this.defaultSiteName = defaultSiteName;
-    }
-
-    @Required
-    public void setContextBuildRetryMaxCount(final int contextBuildRetryMaxCount) {
         this.contextBuildRetryMaxCount = contextBuildRetryMaxCount;
-    }
-
-    @Required
-    public void setContextBuildRetryWaitTimeBase(final long contextBuildRetryWaitTimeBase) {
         this.contextBuildRetryWaitTimeBase = contextBuildRetryWaitTimeBase;
-    }
-
-    @Required
-    public void setContextBuildRetryWaitTimeMultiplier(final int contextBuildRetryWaitTimeMultiplier) {
         this.contextBuildRetryWaitTimeMultiplier = contextBuildRetryWaitTimeMultiplier;
-    }
-
-    @Required
-    public void setModePreview(final boolean modePreview) {
         this.modePreview = modePreview;
-    }
-
-    @Required
-    public void setWatcherPaths(final String[] watcherPaths) {
         this.watcherPaths = watcherPaths;
-    }
-
-    @Required
-    public void setWatcherIgnorePaths(final String[] watcherIgnorePaths) {
         this.watcherIgnorePaths = watcherIgnorePaths;
+        this.watcherCounterLimit = watcherCounterLimit;
+        this.watcherIntervalPeriod = watcherIntervalPeriod;
     }
 
-    @Required
-    public void setWatcherCounterLimit(final int watcherCounterLimit) {
-        this.watcherCounterLimit = watcherCounterLimit;
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
     }
 
-    @Required
-    public void setWatcherIntervalPeriod(final int watcherIntervalPeriod) {
-        this.watcherIntervalPeriod = watcherIntervalPeriod;
+    @Autowired
+    public void setEntitlementValidator(@Lazy final EntitlementValidator entitlementValidator) {
```

Comment:
@sumerjabri We use the `@Autowired` and `@Lazy` combination as a replacement for `@Required` in the case of using a constructor is not possible (due to circular dependencies). This will help avoid the circular issue. I think we have been using this pattern several places in the code in 4.2.0. For example https://github.com/craftercms/engine/commit/c99229229afd74a756f36054429cc4c291887c3e#diff-f9d486f269834a51dfdfa1765a3ac508bf055c45116aa7a9c1cbb6764fdcdd45